tests/conftest.py

```diff
@@ -1,3 +1,15 @@
+# 🌶️🌶️🌶️ Hack to allow testing of both engines
+import os
+
+# If `VLLM_USE_V1=1` is set upon first vLLM import, then there are side-effects
+# in vLLM that will cause the V1 engine to always be used even if our tests
+# update this variable later before constructing each engine.
+# Deleting it here before importing vLLM allows us to continue testing both
+# engines.
+if "VLLM_USE_V1" in os.environ:
+    del os.environ["VLLM_USE_V1"]
```

Comment:
Just wondering if there's something specific to track in vLLM for when to remove this? Related to [this](https://github.com/vllm-project/vllm/blob/main/vllm/engine/llm_engine.py#L2148-L2150), right?

Reply:
Yeah, though it sounds like that was intentionally left in so that code that directly constructs an `AsnycLLMEngine` with `VLLM_USE_V1=1` set will still work.

We could try to go change that in a way that still allows that to work without overriding the class entirely, but I dunno if that's worth the effort if we're only focusing on V1 here anyway. We should talk about what the future of our v0 impl is

Reply:
Updated the comment with a link to the offending vLLM code and an explanation

Reply:
Sounds good, and thanks for adding info to the comment 👍